internal/events/events.go

```diff
@@ -126,14 +126,25 @@ func eventString(writer io.Writer, event flow.Event) {
 }
 
 func printField(writer io.Writer, field cadence.Field, value cadence.Value) {
+	v := value.String()
 	var typeId string
+
+	defer func() {
+		if err := recover(); err != nil {
+			v = fmt.Sprintf("%s\n\t\thex: %x", v, v)
+			_, _ = fmt.Fprintf(writer, "\t\t- %s (%s): %s \n", field.Identifier, "?", v)
+		}
+	}()
```

Comment:
I think using recover is more appropriate here because checking if the type is optional doesn't distinguish between [String]? and String?, and we can't check if it is array of string or just string too. In addition, this handles other cases like dictionary

Reply:
If you can't further define the type then recover is fine, but I wouldn't output anything in the recover since it doesn't provide much information to the user.

Reply:
we don't want to print the values for parsed array? 
Currently with output in recover we get
````
Events:          
    Index       0
    Type        A.f8d6e0586b0a20c7.HelloWorld.FooEvent
    Tx ID       5930133c65ee572bf4fd18ea0b2c0024680566d8a5a969c7db151810fa909af5
    Values
                - x (?): ["Event"] 

````
Without output we will get 
````
Events:          
    Index       0
    Type        A.f8d6e0586b0a20c7.HelloWorld.FooEvent
    Tx ID       5930133c65ee572bf4fd18ea0b2c0024680566d8a5a969c7db151810fa909af5
    Values

````

Reply:
Oh nevermind, I thought the deffer still includes the error output as well. We want to get the values ofc, but it would be best if this output is uniform with the normal flow output and possibly extracted into a method, I would however drop the hex value output